src/io/flutter/run/FlutterRunConfigurationProducer.java

```diff
@@ -51,6 +52,12 @@ public boolean isConfigurationFromContext(final @NotNull FlutterRunConfiguration
     return dartFile != null && dartFile.getPath().equals(configuration.getRunnerParameters().getFilePath());
   }
 
+  @Override
+  public boolean shouldReplace(ConfigurationFromContext self, ConfigurationFromContext other) {
+    // Prefer Flutter runner to plain Dart runner for Flutter code.
+    return other.getConfigurationType().getId().equals("DartCommandLineRunConfigurationType");
```

Comment:
I wanted to avoid any possible breakage if the Dart plugin classes change. (How do we handle dependencies like that?)